datastream-client/src/main/java/com/linkedin/datastream/BaseRestClientFactory.java

```diff
@@ -167,9 +167,12 @@ public Thread newThread(Runnable runnable) {
   private void initHttpClientFactory() {
     ThreadFactory eventLoopThreadFactory = new DaemonNamedThreadFactory("R2 Nio Event Loop");
     ThreadFactory schedExecThreadFactory = new DaemonNamedThreadFactory("R2 Netty Scheduler");
+    ThreadFactory callbackThreadFactory = new DaemonNamedThreadFactory("R2 Callback");
+
     _httpClientFactory = new HttpClientFactory.Builder()
         .setNioEventLoopGroup(new NioEventLoopGroup(0, eventLoopThreadFactory))
         .setScheduleExecutorService(Executors.newSingleThreadScheduledExecutor(schedExecThreadFactory))
+        .setCallbackExecutor(Executors.newSingleThreadExecutor(callbackThreadFactory))
```

Comment:
Is a single thread executor sufficient, or should we make this a fixed thread pool of some size?